lib/ansible/runner.rb

```diff
@@ -8,13 +8,25 @@ def run(env_vars, extra_vars, playbook_path)
       private
 
       def run_via_cli(env_vars, extra_vars, playbook_path)
-        result = AwesomeSpawn.run!(ansible_command, :env => env_vars, :params => [{:extra_vars => JSON.dump(extra_vars)}, playbook_path])
+        prepare_tmp_structure
+
+        result = AwesomeSpawn.run!(ansible_command, :env => env_vars, :params => [{:cmdline => "--extra-vars '#{JSON.dump(extra_vars)}'", :playbook => playbook_path}])
         JSON.parse(result.output)
       end
 
+      def prepare_tmp_structure
+        base_dir = "/tmp/ansible-runner/"
+        mkdir(base_dir)
```

Comment:
Would it make sense to use `Dir.mktmpdir` here and just delete the whole directory after we're done?

Reply:
Maybe? Not sure, I was thinking to just put it to /tmp and the OS will take care of cleaning it.